src/services/jobs/jobs.api.ts

```diff
@@ -0,0 +1,46 @@
+import { createApi, fetchBaseQuery } from "@reduxjs/toolkit/dist/query/react";
+import { RootState } from "store/store";
+
+interface Skills {
+  id: number;
+  name: string;
+}
+
+interface Job {
+  title: string;
+  duration: number;
+  durationType: string;
+  rate: number;
+  isDraft: boolean;
+  englishLevel: string;
+  jobDescription: string;
+  createdAt: string;
+  skills: Skills[];
+}
+
+interface QueryParam {
+  id: number;
+}
+
+const jobsApi = createApi({
+  baseQuery: fetchBaseQuery({
+    baseUrl: process.env.REACT_APP_API_URL,
+    prepareHeaders: (headers, { getState }) => {
+      const token = (getState() as RootState).user.authToken;
+      if (token) headers.set("Authorization", `Bearer ${token}`);
+      return headers;
+    },
+  }),
+  endpoints: (builder) => ({
+    fetchDetails: builder.query<Job, QueryParam>({
+      query: (path) => ({
+        url: `job-post/${path.id}`,
```

Comment:
maybe it's better to keep endpoints paths in one place, I think it should be in .env